pkg/templates/encryption.go

```diff
@@ -169,6 +173,8 @@ func (t *TemplateResolver) processEncryptedStrs(options *ResolveOptions, templat
 		return templateStr, nil
 	}
 
+	templateResult.HasSensitiveData = true
```

Comment:
Is this catching general use of the protect function which could be used with any type of resource?

Reply:
@gparvin right, since on the decryption side, it's just looking for any string with the `$ocm_protected:` prefix.